src/main_controller.ts

```diff
@@ -369,20 +368,21 @@ export class MainController implements vscode.Disposable {
     private async validateNvimRuntime() {
         // WEIRD BUT TRUE: $VIMRUNTIME may be inaccessible even though Nvim itself is runnable! #1815
         const luaCode = `
-            return vim.env.VIMRUNTIME
+          local ok = pcall(function()
+            local dir = vim.loop.fs_opendir(vim.env.VIMRUNTIME)
```

Comment:
why not use `vim.fs` ?

Reply:
I tried this. This function will not report an error even if there is no permission. I don't know how to judge whether the operation is successful.

Reply:
Not familiar with these APIs

Reply:
Just a suggestion. But if `vim.fs` returns zero files, then we can treat that as an error because $VIMRUNTIME must never be empty.

Note that `vim.fs.dir` returns an iterator.